jwst/resample/resample_utils.py

```diff
@@ -303,3 +303,28 @@ def decode_context(context, x, y):
         )
 
     return idx
+
+
+def _xy_range_from_bbox(xmin, xmax, ymin, ymax, shape, bbox=None):
+
+    if (xmin is xmax is ymin is ymax) and xmin in (0, None):
+        xmin = ymin = 0
+        ymax, xmax = np.subtract(shape, 1, dtype=int)
+
+        if bbox:
+            ((x1, x2), (y1, y2)) = bbox
+            if x1 > x2:
+                x1, x2 = x2, x1
+            if y1 > y2:
+                y1, y2 = y2, y1
+
+            # use 0.5+/-eps to avoid having edge pixels with flux coming from
+            # a tiny part of input pixels (with low weight). eps can be set
+            # safely to 0.
+            eps = 0.000001
+            xmin = max(0, int(x1 + (0.5 + eps)))
+            ymin = max(0, int(y1 + (0.5 + eps)))
```

Comment:
Is the goal of the above two lines to compute the center of the pixel? If so you may want to use https://github.com/spacetelescope/gwcs/blob/master/gwcs/utils.py#L54

In general there's no guarantee that the bounding_box values use +/- 0.5 of the center of the pixel.

I am curious - does drizzle require the center of the pixels to be used?

Reply:
> I am curious - does drizzle require the center of the pixels to be used?

Used for what? Not as far as I know.

> In general there's no guarantee that the bounding_box values use +/- 0.5 of the center of the pixel.

That's OK.

> Is the goal of the above two lines to compute the center of the pixel? If so you may want to use https://github.com/spacetelescope/gwcs/blob/master/gwcs/utils.py#L54

No. The goal is to compute the first and last row and column that includes the bounding box.